Gordon360/AuthorizationFilters/StateYourBusiness.cs

```diff
@@ -192,6 +193,25 @@ private bool canReadOne(string resource)
 
                         return false;
                     }
+                case Resource.HOUSING:
+                    {
+                        // The members of the apartment application can only read their application
+                        var housingService = new HousingService(new UnitOfWork());
+                        if (housingService.CheckIfHousingAdmin(user_id))
+                        {
+                            return true;
+                        }
+                        else
+                        {
+                            var sess_cde = Helpers.GetCurrentSession().ToString();
+                            int? applicationID = housingService.GetApplicationID(user_id, sess_cde);
+                            if (applicationID != null)
+                            {
+                                return true;
+                            }
```

Comment:
This only ensures that the user is on _some_ application, not that they're on the one they want to read.
```suggestion
                            var sess_cde = Helpers.GetCurrentSession().ToString();
                            int? applicationID = housingService.GetApplicationID(user_id, sess_cde);
                            var requestedApplicationID = (int) context.ActionArguments["applicationID"];
                            if (applicationID != null && applicationID == requestedApplicationID)
                            {
                                return true;
                            }
```

That might not be exactly the corrext syntax but it should be close enough to make it work. There are other examples in StateYourBusiness

Reply:
I agree with this suggestion, this is the intended behavior